submissions/madmaxWMFU/Friends-App/js/app.js

```diff
@@ -0,0 +1,100 @@
+                    </div>
+                    <div class="user-card__cnt">
+                        <div class="profile-card__name"><span>${user.name.first}</span></div>
+                        <div class="profile-card__name"><span>${user.name.last}</span></div>
+                        <div class="profile-card__name"><span>${user.location.city}, ${user.nat}</span></div>
+                        <div class="profile-card__name"><address>Place: ${user.location.street.name}, ${user.location.street.number}</address></div>
+                    </div> 
+                    <span class="user-card__age">${user.dob.age}</span>
+                </div>`;
+    friendZone.insertAdjacentHTML("beforeEnd", temp);
+}
+
+const cleanFriendsZone = () => friendZone.innerHTML = "";
+
+const renderFriendsList = (allFriends) => allFriends.forEach(friend => drawFriendsCards(friend));
+
+const funcFilterList = {
+    filterByGender(list, {gender}) { return gender !== "all" ? list.filter((person) => person.gender === gender) : allFriends.currentList},
+    filterByAge(list, {ageTill, ageTo}) { return list.filter((person) => ageTill <= person.dob.age && person.dob.age <= ageTo)},
+    filterByName(list, {name}) { return name != "" ? list.filter((person) => new RegExp(name, 'i').test(person.name.first)) : null}
+}
+
+const getFilterList = () => {
+    allFriends.changeList = [...allFriends.currentList];
+    const searchValue = {
+        name: document.querySelector(".search-input-name").value || "",
+        gender: document.querySelector("[name='search-input-gender']:checked").value || "all",
+        ageTill: document.querySelector(".search-input-age1").value || allFriends.minAge,
+        ageTo: document.querySelector(".search-input-age2").value || allFriends.maxAge
+    };
+    let funcName = ["filterByName", "filterByGender", "filterByAge"];
+
+    funcName.forEach((func) => {
+        let newArray = funcFilterList[func](allFriends.changeList, searchValue);
+        if(newArray != null) allFriends.changeList = newArray;
+    });
+
+    cleanFriendsZone();
+    renderFriendsList(allFriends.changeList);  
+}
+
+const getSortList = ({target}) => {
+    switch(target.value) {
+        case "nameAsc":
+            allFriends.changeList = allFriends.changeList.sort((a, b) => a.name.first < b.name.first ? -1 : 1);
+            break;
+        case "nameDesc":
+            allFriends.changeList = allFriends.changeList.sort((a, b) => a.name.first > b.name.first ? -1 : 1);
+            break;
+        case "ageAsc":
+            allFriends.changeList = allFriends.changeList.sort((a, b) => a.dob.age - b.dob.age);
+            break;
+        case "ageDesc":
+            allFriends.changeList = allFriends.changeList.sort((a, b) => b.dob.age - a.dob.age);
+            break;
+    }
+    cleanFriendsZone();
+    renderFriendsList(allFriends.changeList);
+};
+
+const resetSearchValues = () => {
+    filters.options[0].selected = true;
```

Comment:
Do not rely on such things because in a such way you are dependent from order of children elements. Children can be reordered and your code will be broken. 

Use selector instead.